discus/util/db.py

```diff
@@ -2,34 +2,43 @@
 
 import pymongo
 import gridfs
+import os
 
 # --- SETTINGS --- #
 
-# TODO - move these to env variables.
-db_username = "test_user"
-db_password = "password1234"
-db_host = "cluster0.5v8p9am.mongodb.net"
+db_username = os.environ['MONGO_USER']
+db_password = os.environ['MONGO_PASS']
+db_host = os.environ['MONGO_HOST']
 
 
 # --- FUNCTIONS --- #
 
-def setup():
+# Connect to the database.
+def setup(attempts=3):
   # Attempt to connect to the database.
-  try:
-    print("connnecting to database...")
-    client = pymongo.MongoClient("mongodb+srv://" + db_username + ":" \
-      + db_password + "@" + db_host + "/")
-    client.server_info()
-    print("connected!")
-  except pymongo.errors.ServerSelectionTimeoutError as err:
-    print("failed to connect")
-    print(err)
+  for attempt in range(0, attempts):
+    try:
+      # Try to connect.
+      if (attempt == 0):
+        print("Attempting to connnect to the database...")
+      else:
+        print("Reattempting to connect to the database...")
+      client = pymongo.MongoClient("mongodb+srv://" + db_username + ":" \
+        + db_password + "@" + db_host + "/")
+      
+      # Check if connected.
+      client.server_info()
+      print("connected!")
```

Comment:
If it connects successfully, we should break out of the loop to keep it from trying to do it again.

Reply:
yeah, that is planned to be added, just not pushed yet